File: backend/src/main/kotlin/app/ehrenamtskarte/backend/graphql/regions/types/Region.kt
package app.ehrenamtskarte.backend.graphql.regions.types

import app.ehrenamtskarte.backend.db.entities.RegionEntity

data class Region(
    val id: Int,
    val prefix: String,
    val name: String,
    val regionIdentifier: String?,
    val dataPrivacyPolicy: String?,
    val activatedForApplication: Boolean,
    val activatedForCardConfirmationMail: Boolean,
    val applicationConfirmationMailNoteActivated: Boolean,
    val applicationConfirmationMailNote: String?,
) {
    companion object {
        fun fromDbEntity(entity: RegionEntity): Region =
            Region(
                entity.id.value,
                entity.prefix,
                entity.name,
                entity.regionIdentifier,
                entity.dataPrivacyPolicy,
                entity.activatedForApplication,
                entity.activatedForCardConfirmationMail,
                entity.applicationConfirmationMailNoteActivated,
                entity.applicationConfirmationMailNote,
            )
    }
}

File: administration/src/routes/user-settings/__mocks__/Region.ts
import { Region } from '../../../../src-gen/graphql'

export const getTestRegion = (options: {
  id?: number
  name?: string
  prefix?: string
  activatedForApplication?: boolean
  activatedForCardConfirmationMail?: boolean
  applicationConfirmationMailNoteActivated?: boolean
  applicationConfirmationMailNote?: string
}): Region => ({
  id: options.id ?? 0,
  name: options.name ?? 'Augsburg',
  prefix: options.prefix ?? 'a',
  activatedForApplication: options.activatedForApplication ?? true,
  activatedForCardConfirmationMail: options.activatedForCardConfirmationMail ?? true,
  applicationConfirmationMailNoteActivated:
    options.applicationConfirmationMailNoteActivated ?? false,
  applicationConfirmationMailNote: options.applicationConfirmationMailNote ?? null,
  dataPrivacyPolicy: null,
  regionIdentifier: null,
})
